src/Lockedfield.php

```diff
@@ -165,7 +165,7 @@ public function getLocks($itemtype, $items_id)
 
         $locks = [];
         foreach ($iterator as $row) {
-            $locks[] = $row['field'];
+            $locks[$row['field']] = $row['value'];
```

Comment:
`getLocks()` is used at many places.
A new method `getLockedValues()` should probably be declared, to prevent changing method signature on a bugfixes version, and ensure there is no BC breaks possible.

Reply:
I'm not sure this BC breaks anything, all is part of main CommonDBTM calls. To me, the `getLocks` method is wrong from the beginning

Reply:
As far as I see, method was legitimate as there is multiple places where we only have to know the locked fields name only, not their values.
The new result is not compatible with the previous one, so if any plugin is already using this method, it will be broken.

Indeed, it is likely that there is actually no plugin that use this method, but we have unwittingly bugged plugins in the past based on this kind of assumption, and unfortunately it sparked a very bitter rhetoric from some plugin developers about us.

Reply:
Pffff... If a plugin tries to override that part, I'd say it's normal it breaks. Seems a real nonsense. Plugins can do everything they want because of old GLPOI crap code, that does not means they should or we should take care of.